qiskit/algorithms/optimizers/spsa.py

```diff
@@ -328,6 +328,9 @@ def _minimize(self, loss, initial_point):
                         k, self.maxiter + 1, time() - iteration_start)
 
             if self.callback is not None:
+                self._nfev += 1
+                fx_next = loss(x_next)
```

Comment:
I think this might be even better: this way we really only evaluate once 🙂 
```suggestion
                # if we didn't evaluate the function yet, do it now
                if not self.blocking:
                    self._nfev += 1
                    fx_next = loss(x_next)
```